### ISSUE TITLE
Multiuser Issue for Android 11 GSI GApps

### ISSUE THREAD

**Issue by edwining01** [phhusson/treble_experimentations#1794]
# TL;DR

I have been using the GSI image for a few days. The device flashed a rom with GApps. When switched to a secondary user, an error message, "This device isn't Play Protect certified", will be promoted as a notification.

- Do any other devices have this problem or is it only on Poco M3?

- Is that some kind of prop of the secondary user being override?

- Is the secondary user missing some permission to read the GSF id?

- What else do I needed to log for further investigation?

---

## Steps

Play store is still quite important for me personally.

At first after flashed the GSI ROM with GApps and after set up the primary user, I got the "This device isn't Play Protect certified" notification.

Then, I [verified](https://support.google.com/android/answer/10248227?hl=en#zippy=%2Cdevice-isnt-certified) my device on Google after a few searching. The warning had been dismissed.

I thought it works fine but problem happened while creating new user:

- The first try was to turn off any internet (wifi and 4G), then create and set up a new user. It seemed worked at first. However after I reboot the device and switch back to that account, warning came back (with internet).

- I then tried create a new user again and reboot and switch back without any network connected. Everything looked fine.

- Another try is to create new user and set up offline, and add new google account in `settings` with internet connection. The warning came back after the process `Checking Info` of `settings: Add account`.

(All three cases above won't affect the primary user.)

Here are the logcat result of the last case. I hope that would be useful.

[logcat files (gist)](https://gist.github.com/edwining01/bf17c9e8315957abb9a38174a1cda5c1)

[logcat_0_switch_user.txt](https://gist.github.com/edwining01/bf17c9e8315957abb9a38174a1cda5c1/raw/554ceb8b4e781ff1fef7581c342349f9c235a2f0/logcat_0_switch_user.txt): Switch user after user created.
[logcat_1_setup_user.txt](https://gist.github.com/edwining01/bf17c9e8315957abb9a38174a1cda5c1/raw/554ceb8b4e781ff1fef7581c342349f9c235a2f0/logcat_1_setup_user.txt): Init setup for user without wifi.
[logcat_2_add_gac.txt](https://gist.github.com/edwining01/bf17c9e8315957abb9a38174a1cda5c1/raw/554ceb8b4e781ff1fef7581c342349f9c235a2f0/logcat_2_add_gac.txt): Tried to add Google account in `settings` after wifi was on.

**Comment by phhusson** [comment]
Well issue is that GSF ID is different for different users, to get it do:

sqlite3  /data/user/11/com.google.android.gsf/databases/gservices.db  'select * from main where name = "android_id";'

the "11" might change. just list folders in /data/user

**Comment by edwining01** [comment]
It works!

Never know there are more than one database before. Thanks a lot! 😄

**Comment by thewh1teagle** [comment]
For anyone who facing the same issue, 
just want to help you - 
if it still doesn't works
**Clear data for both Play Store and Play Services, then reboot. do this even twice**
